pkg/reconciler/service/service.go

```diff
@@ -540,12 +540,23 @@ func updateStageTargetRevisions(ro *v1.RolloutOrchestrator, config *RolloutConfi
 	var stageRevisionTarget []v1.TargetRevision
 	if !targetsEqual(ro.Spec.InitialRevisions, ro.Spec.TargetRevisions) {
 		startRevisions := getStartRevisions(ro)
-		if len(startRevisions) == 0 || config.OverConsumptionRatio >= common.HundredPercent {
+		if len(startRevisions) == 0 {
 			// If the index is out of bound, assign the StageTargetRevisions to the final TargetRevisions.
 			ro.Spec.StageTargetRevisions = append([]v1.TargetRevision{}, ro.Spec.TargetRevisions...)
 			return nil
 		}
 
+		if config.OverConsumptionRatio >= common.HundredPercent {
```

Comment:
Where do we check the resourceUtil mode?

Reply:
The strategy is set here into the rolloutOrchestrator: https://github.com/knative-extensions/serving-progressive-rollout/blob/main/pkg/reconciler/service/service.go#L375

Based on different rollout strategy, different implementations of the Reconcile func will be called:
https://github.com/knative-extensions/serving-progressive-rollout/blob/main/pkg/reconciler/rolloutorchestrator/rolloutorchestrator.go#L88